### DISCUSSION TITLE
How to detect circular dependencies within a pnpm monorepo

### DISCUSSION THREAD

**Discussion by vorillaz** [pnpm/pnpm#5544] [Q&A]
Hi there,
We are currently using pnpm inside a monorepo building and bundling each app independently.
Our next.js app is having a hard time during development, there is a memory leak somewhere (related issue https://github.com/vercel/next.js/issues/32314) and I am trying to spot the culprit.

Is there any way to detect circular dependencies during bundling for a pnpm monorepo?
I used [madge](https://github.com/pahen/madge) and fixed a few spots but it does not work properly.
Thanks in advance.

**Comment by douglasg14b** [comment]
:rip: this question I guess :/

#1 on search results.

Did you find out some good options here?
